### PR TITLE
Add holes support for geojson

### LINKED ISSUE #28: Multipolygon holes
Thanks for this awesome tool. Is there something specific to make sure holes are also drawn?

image

### PR DESCRIPTION
**Pull request by zifeo**
Fix #28 

Capture d’écran 2019-06-15 à 02 41 21

@PatrikHlobil

### CODE REVIEW

**Comment by PatrikHlobil** [comment]
@zifeo: wow, that was fast. Thanks for your help 😁.

Did you run the tests so far? Can I help you run the tests if you haven’t?

I try to give you feedback and merge the request in the next few days 😉. 

I just started adding automatic azure pipeline tests and will soon include them in den master branch. Then we can automatically trigger azure tests at every commit 👌🏼.

Thanks again, without the community support it is hard to improve such projects 😉.

Best Patrik

**Comment by zifeo** [comment]
I run `pytest` on on tests for pandas and geopandas without issue, I do not have a spark setup right now. Is there something more I can do?

**Comment by PatrikHlobil** [comment]
That's fine, the pyspark test is rather trivial and does not interfer with the geoplots that you changed. 

If you have time, it would be cool if you could:

1.) add a test for the hole geoplots (just use your ipynb example). The tests are very simple so far, I just basically check if a plot could successfully be created, since I haven't found a nice and easy way to compare bokeh plots (the JSON representation has random ids, which makes it rather complicated). So, just add a test_hole_geplot() function to the test_GeoPandasBokeh.py file. You can just copy and modify the following example:
```python 
def test_geolayers_simple():
    "Tests for simple"

    # Read in GeoJSON from URL:
    df_states = gpd.read_file(
        r"https://raw.githubusercontent.com/PatrikHlobil/Pandas-Bokeh/master/Documentation/Testdata/states/states.geojson"
    )

    figure = df_states.plot_bokeh(simplify_shapes=10000, show_figure=False)

    with open(  # +ä
        os.path.join(directory, "Plots", "Geolayers_Simple.html"), "w"
    ) as f:
        f.write(pandas_bokeh.embedded_html(figure))

    assert True
```

2.) Rename the shapes.geojson (maybe to hole_shapes.geojson) and move it to the folder with the testdata (so then e.g. in Pandas-Bokeh/Documentation/Testdata/hole_shapes/hole_shapes.geojson). 

3.) Delete the GeoJSON.ipynb

Best, Patrik

**Comment by zifeo** [comment]
@PatrikHlobil Done. I would suggest to compare images to ensure there is no regression.

File: Tests/test_GeoPandasBokeh.py [hunk 1]
```diff
@@ -97,5 +97,20 @@ def test_geolayers_slider():
...
 
-
 #     assert False  # ASSERT IS FALSE BECAUSE SLIDER IS NOT VISIBLE RIGHT NOW:
+
+
+def test_hole_geplot():
+    "Tests for (multi-)polygones with holes."
+
+    df = gpd.GeoDataFrame.from_file(
+        os.path.join(directory, '../Documentation/Testdata/hole_shapes/hole_shapes.geojson')
```
Comment (PatrikHlobil) [inline Tests/test_GeoPandasBokeh.py]: Use test_set_directory here

**Comment by zifeo** [comment]
@PatrikHlobil Missed that one, corrected!

**Comment by PatrikHlobil** [comment]
Perfect, I can merge now. But before I do so, do you want to add your name to the CONTRIBUTORS.md file?

**Comment by PatrikHlobil** [comment]
@zifeo : I will now merge your PR and add your name to the contributors.md file ;). Thanks for your big help, 

cheers Patrik

**Pull request merged**